### PR TITLE
Vendoring golint and updating check-golint.sh to use that

### PR DESCRIPTION
**Pull request by nikhiljindal**
There is no good way to just add one vendor dependency: https://github.com/Masterminds/glide/issues/321
Hence the big update.

Autogenerated by running `glide get golang.org/x/lint/golint --strip-vendor`

cc @G-Harmon

### CODE REVIEW

**Comment by nikhiljindal** [comment]
Added another commit to update check-golint.sh to use golint from vendor.

Lets see if the tests pass now

File: scripts/check-golint.sh [hunk 1]
```diff
@@ -18,6 +18,14 @@ set -e
 set -u
 set -o pipefail
 
+# Set a GOBIN to install golint.
+ROOT_DIR="$(cd "$(dirname "${BASH_SOURCE}")/.." && pwd -P)"
```
Comment (nikhiljindal) [inline scripts/check-golint.sh]: Code inspired from https://github.com/kubernetes/kubernetes/blob/master/hack/update-bazel.sh#L30 as suggested by @BenTheElder and @@ixdy

**Comment by nikhiljindal** [comment]
The tests ran and found some errors.
Pushed 2 new commits to fix build errors and vet errors.

**Comment by nikhiljindal** [comment]
The test is now failing with:
```
W0407 01:32:49.082] vendor/golang.org/x/lint/lint.go:26:2: cannot find package "golang.org/x/tools/go/ast/astutil" in any of:
W0407 01:32:49.083] 	/go/src/github.com/GoogleCloudPlatform/k8s-multicluster-ingress/vendor/golang.org/x/tools/go/ast/astutil (vendor tree)
W0407 01:32:49.083] 	/usr/local/go/src/golang.org/x/tools/go/ast/astutil (from $GOROOT)
W0407 01:32:49.083] 	/go/src/golang.org/x/tools/go/ast/astutil (from $GOPATH)
W0407 01:32:49.084] vendor/golang.org/x/lint/lint.go:27:2: cannot find package "golang.org/x/tools/go/gcexportdata" in any of:
W0407 01:32:49.084] 	/go/src/github.com/GoogleCloudPlatform/k8s-multicluster-ingress/vendor/golang.org/x/tools/go/gcexportdata (vendor tree)
W0407 01:32:49.084] 	/usr/local/go/src/golang.org/x/tools/go/gcexportdata (from $GOROOT)
W0407 01:32:49.084] 	/go/src/golang.org/x/tools/go/gcexportdata (from $GOPATH)
```

I can try explicit `glide get` for these packages but not sure why didnt `glide get golint` fetch them as well

**Comment by nikhiljindal** [comment]
All tests passed! :)

**Comment by G-Harmon** [comment]
/LGTM
There are still 5 commits in this PR. I thought you wanted to distill this down into 2 or 3? Feel free to merge after answering or squashing.

**Comment by nikhiljindal** [comment]
Verified that the ci job is green as well: https://k8s-testgrid.appspot.com/sig-multicluster-kubemci#ci-kubemci-unit-tests

File: scripts/check-golint.sh [hunk 2]
```diff
@@ -18,6 +18,14 @@ set -e
...
+export GOBIN="${ROOT_DIR}/_output/local/bin"
```
Comment (ixdy) [inline scripts/check-golint.sh]: you may want to git ignore the `_output` directory, too.

Comment (BenTheElder) [reply scripts/check-golint.sh]: @nikhiljindal definitely want to make sure you're doing this, I don't see any .gitignore currently

Comment (G-Harmon) [reply scripts/check-golint.sh]: I kind of think this is the first PR that exhibits degraded coverage data, #190. Perhaps due to changing the path? Does this PR change where/what "go coverage" is executed?

Comment (nikhiljindal) [reply scripts/check-golint.sh]: Sent https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/pull/191 to add .gitignore as per comments here.

**Comment by BenTheElder** [comment]
Are you getting coverage data for vendor or golint as well now?

**Pull request merged**